exporters/prometheus/CMakeLists.txt

```diff
@@ -31,6 +31,25 @@ target_link_libraries(
   opentelemetry_exporter_prometheus
   PUBLIC opentelemetry_metrics prometheus-cpp::pull prometheus-cpp::core)
 
+# We create a separate target for utils library
+add_library(
+  opentelemetry_exporter_prometheus_utils
+  src/exporter_utils.cc)
+
+set_target_properties(opentelemetry_exporter_prometheus_utils
+                      PROPERTIES EXPORT_NAME prometheus_exporter_utils)
+set_target_version(opentelemetry_exporter_prometheus_utils)
+
+target_include_directories(
+  opentelemetry_exporter_prometheus_utils
+  PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
+         "$<INSTALL_INTERFACE:include>")
+
+list(APPEND PROMETHEUS_EXPORTER_TARGETS opentelemetry_exporter_prometheus_utils)
+target_link_libraries(
+  opentelemetry_exporter_prometheus_utils
+  PUBLIC opentelemetry_metrics prometheus-cpp::core)
+
```

Comment:
Targets are also added in cmake config - https://github.com/open-telemetry/opentelemetry-cpp/blob/d036d8227f6f1f988c611aa6a0312f3ccd346d63/cmake/opentelemetry-cpp-config.cmake.in#L77 But I am not sure of the implications in that case - as the symbols from exporter_utils.cc would be part of two target belonging to same package `OPENTELEMETRY_CPP_LIBRARIES`. Will let @owent comment on that :)

Reply:
I don't think that should be an issue, resolving it. Sorry for confusion :)

Reply:
I think we should remove `src/exporter_utils.cc` from and link `opentelemetry_exporter_prometheus_utils` into `opentelemetry_exporter_prometheus` if we want to split this file into a standalone target.Or there may be conflict when some components link both `opentelemetry_exporter_prometheus_utils` and `opentelemetry_exporter_prometheus`.

BTW: The documents and the component name for `OPENTELEMETRY_CPP_LIBRARIES` should also be added into `cmake/opentelemetry-cpp-config.cmake.in`.